### PR TITLE
made it so past moth starts from month start. implemented past year

### PR DESCRIPTION
**Pull request by rt-kill**

### CODE REVIEW

File: cole_bot.py [hunk 1]
```diff
@@ -123,3 +123,18 @@ async def get_member_count(data):
 if __name__ == "__main__":
     my_bot.ipc.start()  # start the IPC Server
     my_bot.run(os.getenv("bot_token"))
+
+
+@my_bot.ipc.route()
+async def user_id_to_username(data):
```
Comment (Ruborcalor) [inline cole_bot.py:129]: this function is already defined, why are you defining it again?

File: common/async_utilities.py [hunk 2]
```diff
@@ -412,7 +412,16 @@ async def get_user_timeseries(redis_client, user_id, time_interval):
             }
         )
 
-    return list(reversed(timeseries))
+    timeseries = list(reversed(timeseries))
+
+    if time_interval == "allTime":
```
Comment (Ruborcalor) [inline common/async_utilities.py:417]: this seems like a slow way of writing this. each `del timeseries[0]` is `O(n)`. Instead find the index of the first non zero value and then take the subarray from that point onwards.

https://wiki.python.org/moin/TimeComplexity
image

**Review by Ruborcalor — CHANGES_REQUESTED** [review]

File: cole_bot.py (hunk 1 above)
Comment (rt-kill) [reply cole_bot.py]: it has been removed in the latest push

File: common/async_utilities.py (hunk 2 above)
Comment (rt-kill) [reply common/async_utilities.py:417]: fixed in the latest push

**Review by Ruborcalor — APPROVED** [review]

**Pull request merged**